pages/docs/adapters/calculix/adapter-calculix-pastix-build.md

```diff
@@ -0,0 +1,166 @@
+
+### Troubleshooting
+
+- On some occasions, a Python script called by CMake generates an incorrect Makefile because of errors in regular expressions. (The script being `cmake_modules/morse_cmake/modules/precision_generator/genDependencies.py`) This should be fixed by calling `pip install regex` (which requires installing the `python3-pip` Ubuntu package). You may also need to replace the `import re` line in that script by `import regex as re`, but the necessity seems to fluctuate among different machines.
+- Some parts of the code require older versions of the GNU compilers. You may have to replace `gcc` by `gcc-7` and similarly for `g++` and `gfortran` in the `make_pastix.sh` script. This requires installing the relevant Ubuntu packages.
+
+## Building ARPACK, a CalculiX dependency
+
+Calculix relies on ARPACK, and when built with PaStiX, we cannot rely on standard distributions of that library, because it doesn't feature 8-bytes integers by default. We need to compile it ourself. The source code can be found [here](https://www.caam.rice.edu/software/ARPACK/), or by running these commands :
+
+```bash
+cd ~
+wget https://www.caam.rice.edu/software/ARPACK/SRC/arpack96.tar.gz
+wget https://www.caam.rice.edu/software/ARPACK/SRC/patch.tar.gz
+zcat arpack96.tar.gz | tar -xvf -
+zcat patch.tar.gz    | tar -xvf -
+
+```
+
+Before building the library, the following modifications are required :
+
+- In `ARmake.inc`, change `PLAT` by the appropriate suffix (the adapter's makefile assumes INTEL)
+- In `ARmake.inc`, change the Fortran compilation flags to add `-fdefault-integer-8`. You may also need to remove the flag `-cg89`.
+- If you extracted the archive on another folder than your home repository, update `home` in `ARmake.inc` accordingly.
+- In the file `UTIL/second.f`, comment with a star the line containing `EXTERNAL ETIME`.
+
+Once all of these are done, simply run `make lib` in the `ARPACK` folder.
+
+## Building the adapter
+
+To build the adapter, clone its repository and checkout the `2.17` branch :
+
+```bash
+    git clone -b v2.17 https://github.com/precice/calculix-adapter
+    cd calculix-adapter
+```
+
+### Fixing the code
+
+Due to some conflicts between CalculiX, PaStiX and the adapter (both CalculiX and PaStiX have a `pastix.h` file, and neither of them is local from the point of view of the adapter), some changes are required in the CalculiX codebase. We provide a script, `pastix_pre_build.sh` that does these changes. Run it.
+
+```bash
+    ./pastix_pre_build.sh
+```
+
+### Compilation
+
+To build the adapter, use the provided `Makefile_i8_PaStiX` : the regular Makefile would build the adapter without PaStiX. Assuming you followed the previous steps, it should be useable without modifications; otherwise, some paths updates could be required. Run this command in the `calculix-adapter` (and be sure to checkout the `2.17` branch) folder :
```

Comment:
I see quite often a space before colons. Any editor-specific reason?

```suggestion
To build the adapter, use the provided `Makefile_i8_PaStiX`: the regular Makefile would build the adapter without PaStiX. Assuming you followed the previous steps, it should be useable without modifications; otherwise, some path updates could be required. Run this command in the `calculix-adapter` folder (and be sure to checkout the `2.17` branch):
```

(I would again remove the specific version and assume that we are already in v2.17 or higher)

Reply:
Space before colon is standard in French, I keep forgetting it's not supposed to be done in English :|

Reply:
oh! I learned something today! :smiley: